client/me/purchases/remove-plan-dialog/style.scss

```diff
@@ -0,0 +1,3 @@
+.non-primary-domain-dialog {
+	max-width: 500px;
+}
```

Comment:
There is a prop called isFullScreen maybe you can utilize that prop here.

https://github.com/Automattic/wp-calypso/blob/e9d2f791cb8f207a10f80e251cec8e309e5d1963/packages/components/src/dialog/index.tsx#L17

Also the guide to style the dialog component can be found in the docs.
https://wpcalypso.wordpress.com/devdocs/packages/components/src/dialog/README.md?term=Dialog